kable-core/src/jsMain/kotlin/Bluetooth.kt

```diff
@@ -1,103 +1,36 @@
 package com.juul.kable
 
-import com.benasher44.uuid.Uuid
-import com.juul.kable.Bluetooth.Availability.Available
-import com.juul.kable.Bluetooth.Availability.Unavailable
-import com.juul.kable.Reason.BluetoothUndefined
-import com.juul.kable.external.BluetoothAvailabilityChanged
-import com.juul.kable.external.BluetoothServiceUUID
-import com.juul.kable.external.RequestDeviceOptions
-import js.objects.jso
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.await
-import kotlinx.coroutines.channels.awaitClose
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.callbackFlow
-import kotlinx.coroutines.flow.onStart
-import org.w3c.dom.events.Event
-import kotlin.js.Promise
-import com.juul.kable.external.Bluetooth as JsBluetooth
+import com.juul.kable.external.Bluetooth
+import com.juul.kable.external.bluetooth
+import js.errors.ReferenceError
+import kotlinx.browser.window
 
-public actual enum class Reason {
-    /** `window.navigator.bluetooth` is undefined. */
-    BluetoothUndefined,
-}
-
-private const val AVAILABILITY_CHANGED = "availabilitychanged"
-
-internal actual val bluetoothAvailability: Flow<Bluetooth.Availability> = callbackFlow {
-    if (safeWebBluetooth == null) return@callbackFlow
-
-    // https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/onavailabilitychanged
-    val listener: (Event) -> Unit = { event ->
-        val isAvailable = event.unsafeCast<BluetoothAvailabilityChanged>().value
-        trySend(if (isAvailable) Available else Unavailable(reason = null))
-    }
-
-    bluetooth.apply {
-        addEventListener(AVAILABILITY_CHANGED, listener)
-        awaitClose {
-            removeEventListener(AVAILABILITY_CHANGED, listener)
-        }
-    }
-}.onStart {
-    val availability = if (safeWebBluetooth == null) {
-        Unavailable(reason = BluetoothUndefined)
-    } else {
-        val isAvailable = bluetooth.getAvailability().await()
-        if (isAvailable) Available else Unavailable(reason = null)
+/**
+ * @return [Bluetooth] object or `null` if bluetooth is [unavailable](https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth#browser_compatibility).
+ */
+internal fun bluetoothOrNull(): Bluetooth? {
+    val navigator = try {
+        window.navigator
+    } catch (e: ReferenceError) {
```

Comment:
It is kinda sad to me that Kotlin doesn't differentiate the Js targets (e.g. browser vs node) and has these definitions in the standard library that just crash if you use them in the wrong environment. 😞

Reply:
Granted it'd crash on node if you tried to access `window.navigator` in pure JS. 🤷

Reply:
Seems sensible enough. I guess it is too late for JB to split the targets so we could build for `js().browser()` and `js().node()`, but it sucks that they provide an API for `window.navigator` that just crashes. 🫠